### PR TITLE
0.3

### PR DESCRIPTION
**Pull request by ratamovic**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,49 +50,120 @@ through ssh.
...
 a toolchain or two...
 
 ```
+# check available cross-compilation targets
+rustup target list
+
 # a few ios devices
 rustup target install aarch64-apple-ios
 rustup target install armv7-apple-ios
 rustup target install arm-linux-androideabi
```
Comment (fredszaq) [inline README.md]: remove android device from here

File: README.md [hunk 2]
```diff
@@ -21,49 +50,120 @@ through ssh.
...
-You'll need the usual ANDROID_NDK_HOME, and `adb` somewhere in your path.
-`fb-adb` will be used if its available on the path, giving better error
-handling.
+You'll need to make a stand-alone toolchain as explained in the [official Android documentation](https://developer.android.com/ndk/guides/standalone_toolchain.html) (Dinghy cannot use the NDK directly), and `adb` somewhere in your path.
+`fb-adb` will be used if its available on the path. It allows dinghy to forward correctly binaries/benches/tests exit codes from the conneced device to the host platform.
 Also, your phone must have developer options enabled.
 
-## iOS simulator setup
+You will also probably have to set up a toolchain: rustc needs a working linker
+to be able to generate executable. On my case, for using on a RaspberryPi, I
```
Comment (fredszaq) [inline README.md]: raspberry pi in section android setup  ??

File: README.md [hunk 3]
```diff
@@ -268,42 +354,170 @@ the_data = { source = "../data-2017-02-05", copy_git_ignored = true }
...
+```
+
+Overlays are usually outside the toolchain sysroot. As a consequence, Dinghy must overrides the `prefix` pkg-config variable to provide a correct overlay path relative to the toolchain sysroot, despite being outside of it.
+Hence, when writing a *.pc* file, it's very *important* to:
+- Define a `prefix` variable that Dinghy can override
+- Consider that this `prefix` points to the root of the overlay directory
+
+#### Overlay runtime
+
+To make sure overlays are available at runtime, during benches, run or tests, Dinghy will copied al the `.so` files linked by the linker script during a build on the target device before running the appropriate executable.
```
Comment (fredszaq) [inline README.md]: all

File: README.md [hunk 4]
```diff
@@ -268,42 +354,170 @@ the_data = { source = "../data-2017-02-05", copy_git_ignored = true }
...
+
+An overlay is like an additional sysroot. So you can also create directories and subdirectories. For example, we use a tensorflow overlay on our android projects with the following structure:
+```
+android-arm64
+|- overlay
+    |- tensorflow
+        |- usr
+            |- include
+                |- pkgconfig
+                    |- mylib.pc
```
Comment (fredszaq) [inline README.md]: mylib when saying it's a tensorflow overlay

Comment (ratamovic) [reply README.md]: k

File: README.md (hunk 3 above)
Comment (ratamovic) [reply README.md]: k

File: README.md (hunk 2 above)
Comment (ratamovic) [reply README.md]: k

File: README.md (hunk 1 above)
Comment (ratamovic) [reply README.md]: k

**Pull request merged**